File: dataRetriever/downloader.js
var fs = require('fs');
var request = require('request');
var sha256 = require('js-sha256').sha256;
var download = require('image-downloader')


const directory = "img/"

var downloadImage = function (uri, filename, callback) {
	const options = {
		  url: uri,
		  dest: filename 
	}
	download.image(options)
		.then(({ filename, image }) => {
			callback();
		})
		.catch((err) => {
			console.error(err)
		})
};

var loadFile = function (filename) {
	var contents = fs.readFileSync(filename);
	return JSON.parse(contents);
}

var saveFile = function (filename, object) {
	var contents = JSON.stringify(object, null, 2);
	fs.writeFile(filename, contents, function(err) {
		if(err)
			console.log(err);
	});
}

var main = function() {

	const inputFile  = "edited.json";
	const outputFile = "output.json";
	var file = loadFile(inputFile);

	console.log(file.intersections[0]);
	for(var i = 0 ; i < file.intersections.length ; i++) {
		
		if(file.intersections[i]["images"] != undefined) {
			for(var j = 0 ; j < file.intersections[i].images.length ; j++) {
				var filename = sha256(file.intersections[i].images[j]); 
				downloadImage(file.intersections[i].images[j], directory + filename + '.jpg', function() {

				});
				file.intersections[i].images[j] = filename + '.jpg';
			}
		}
		else {
			console.log(i);
		}
	}

	saveFile(outputFile, file);

}

main();

File: dataRetriever/downloader.py
import json
import hashlib
import requests
import os.path


def downloadImage(url, filename):
    if not os.path.exists('img/' + filename):
        with open('img/' + filename, 'wb') as f:
            f.write(requests.get(url).content)

def main():

    input = json.loads(open('edited.json').read())

    print(input)

    for inter in input['intersections']:
        for direction in inter['direction']:
            for image in direction['images']:
                image['id'] = hashlib.sha256(image['pathExt'].encode('utf-8')).hexdigest()
                image['pathInt'] = 'img/' + image['id'] + '.jpg'
                downloadImage(image['pathExt'], image['id'] + '.jpg')

    with open('output2.json', 'w') as outfile:
        json.dump(input, outfile)


if __name__ == "__main__":
    main()
